### PR TITLE
feat: Add 'move'-repository for rest-api

### PR DESCRIPTION
**Pull request by andpopov**
close: #1141 
@olenagerasimova @dgarus Please review

### CODE REVIEW

File: src/main/java/com/artipie/api/RepositoryRest.java [hunk 1]
```diff
@@ -205,4 +205,34 @@ private void removeRepo(final RoutingContext context) {
...
+        final SettingsDuplicatesVerifier duplicates =
+            new SettingsDuplicatesVerifier(rname, this.crs);
+        final ExistenceVerifier existence = new ExistenceVerifier(rname, this.crs);
+        final Validator validator = validator(
+            validator(reserved::valid, reserved::message, HttpStatus.BAD_REQUEST_400),
+            validator(existence::valid, existence::message, HttpStatus.NOT_FOUND_404),
+            validator(duplicates::valid, duplicates::message, HttpStatus.CONFLICT_409)
+        );
+        if (validator.validate(context)) {
+            final String newname = BaseRest.readJsonObject(context).getString("name");
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/api/RepositoryRest.java:225]: I'd suggest to call json field for new name `new_name` as it was in [previous API](https://github.com/artipie/front#repositories-api)

File: src/main/java/com/artipie/api/BaseRest.java [hunk 2]
```diff
@@ -41,6 +44,17 @@ protected Handler<RoutingContext> errorHandler(final int code) {
...
+     * Read body as JsonObject.
+     * @param context RoutingContext
+     * @return JsonObject
+     */
+    protected static JsonObject readJsonObject(final RoutingContext context) {
+        return (JsonObject) (Json.createReader(
+            new StringReader(context.body().asString())
+        ).read());
+    }
+
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/api/BaseRest.java:57]: sorry, I've missed this static `BaseRest#validator(com.artipie.api.Validator...)` method which can be easily remade into normal object - decorator for `Validator`. The class can be a subclass of `Validator` and called `All`. Here is an [example](https://github.com/artipie/http/blob/45cc8611213da923e2e35ada3e8dfb676214e5ab/src/main/java/com/artipie/http/rt/RtRule.java#L68) from http, please, follow it.

File: src/main/java/com/artipie/api/RepositoryRest.java [hunk 3]
```diff
@@ -205,4 +205,34 @@ private void removeRepo(final RoutingContext context) {
...
+            final RepositoryName newrname;
+            if (new Layout.Flat().toString().equals(this.layout)) {
+                newrname = new RepositoryName.Flat(newname);
+            } else {
+                newrname = new RepositoryName.Org(newname, context.pathParam(RepositoryName.UNAME));
+            }
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/api/RepositoryRest.java:231]: it seems, that new repository name should be validated the way as the old one...

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@andpopov thanks, added several comments...

File: src/main/java/com/artipie/api/RepositoryRest.java (hunk 3 above)
Comment (andpopov) [reply src/main/java/com/artipie/api/RepositoryRest.java:231]: @olenagerasimova thanks, I have added changes, please review

**Review by olenagerasimova — APPROVED** [review]
@andpopov thanks

**Pull request merged**